docs/references/config/config.toml.md

```diff
@@ -0,0 +1,1749 @@
+priv_validator_state_file = "data/priv_validator_state.json"
+```
+
+| Value type          | string                                          |
+|:--------------------|:------------------------------------------------|
+| **Possible values** | relative directory path, appended to `$CMTHOME` |
+|                     | absolute directory path                         |
+
+The default relative path translates to `$CMTHOME/data/priv_validator_state.json`. In case `$CMTHOME` is unset, it
+defaults to `$HOME/.cometbft/data/priv_validator_state.json`.
+
+### priv_validator_laddr
+TCP or UNIX socket listen address for CometBFT that allows external consensus signing processes to connect.
+```toml
+priv_validator_laddr = ""
+```
+
+| Value type          | string                                                |
+|:--------------------|:------------------------------------------------------|
+| **Possible values** | TCP Stream socket (`"tcp://127.0.0.1:26658"`)         |
+|                     | Unix domain socket (`"unix:///var/run/privval.sock"`) |
+
+When consensus signing is outsourced from CometBFT (typically to a Hardware Security Module, like a
+[YubiHSM](https://www.yubico.com/product/yubihsm-2) device), this address is opened by CometBFT for incoming connections
+from the signing service.
+
+Make sure the port is available on the host machine and firewalls allow the signing service to connect to it.
+
+More information on a supported signing service can be found in the [TMKMS](https://github.com/iqlusioninc/tmkms)
+documentation.
+
+### node_key_file
+Path to the JSON file containing the private key to use for node authentication in the p2p protocol.
+```toml
+node_key_file = "config/node_key.json"
+```
+
+| Value type          | string                                          |
+|:--------------------|:------------------------------------------------|
+| **Possible values** | relative directory path, appended to `$CMTHOME` |
+|                     | absolute directory path                         |
+
+The default relative path translates to `$CMTHOME/config/node_key.json`. In case `$CMTHOME` is unset, it defaults to
+`$HOME/.cometbft/config/node_key.json`.
+
+### abci
+The mechanism used to connect to the ABCI application.
+````toml
+abci = "socket"
+````
+
+| Value type          | string     |
+|:--------------------|:-----------|
+| **Possible values** | `"socket"` |
+|                     | `"grpc"`   |
+
+This mechanism is used when connecting to the ABCI application over the [proxy_app](#proxy_app) socket.
+
+### filter_peers
+When connecting to a new peer, filter the connection through an ABCI query to decide, if the connection should be kept.
+```toml
+filter_peers = false
+```
+
+| Value type          | boolean |
+|:--------------------|:--------|
+| **Possible values** | `false` |
+|                     | `true`  |
+
+When this setting is `true`, the ABCI application has to implement a query that returns `true` or `false` depending on
```

Comment:
It actually returns an error:

https://github.com/cometbft/cometbft/blob/main/node/setup.go#L379-L415